app/views/shared/_step_indicator.html.erb

```diff
@@ -0,0 +1,28 @@
+<%#
+yields: Steps markup.
+locals:
+* class: Extra CSS classes to apply to wrapper.
+* steps: Hash of step key to step partial data.
+* current_step: Current step.
+%>
+<%
+classes = ['step-indicator']
+classes << local_assigns[:class] if local_assigns[:class]
+content = yield.presence || begin
+  current_step_index = steps.index { |step| step[:slug] == local_assigns[:current_step] }
+  steps.each_with_index.reduce(''.html_safe) do |result, (step, index)|
+    result += render(
```

Comment:
the `''.html_safe` and the `result += render` are very uncommon patterns to me.... I *think* we could just do normal ERB like this:

```erb
<% block_content = yield %>

<%= tag.div class: classes do %>
  <% if block_content.present? %>
    <%= block_content %>
  <% else %>
    <% steps.each_with_index do |step, index| %>
      <%= render ... %>
    <% end %>
  <% end %>
<% end %>
```

Reply:
Yeah, that seems nicer. Updated in 0eb2962. I've another one of these in `layouts/flow_step.html.erb` I'll try to rethink on Monday.